app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java

```diff
@@ -4446,90 +4458,74 @@ public Boolean UploadOfflineFeedbackRenewal(final String ActivityName) {
 
             return false;
         }
+
         if (!general.isNetworkAvailable(mContext)) {
             ShowDialog(mContext.getResources().getString(R.string.NoInternet));
             return false;
         }
-        final ProgressDialog pd;
+
         pd = ProgressDialog.show(mContext, "", mContext.getResources().getString(R.string.Uploading));
 
         new Thread() {
             public void run() {
-                for (int i = 0; i < JSONfiles.length; i++) {
-                    UploadCounter = i + 1;
-                    //runOnUiThread(ChangeMessage);
-
-                    String JsonFileName = getFeedRenewalText(JSONfiles[i].getName());
-
-                    JSONObject obj = new JSONObject();
-
-                    ToRestApi rest = new ToRestApi();
-
-                    Boolean serv = null;
-                    int resInt = 2;
-
-                    if (ActivityName.equals("renewal")) {
-                        try {
-                            obj = new JSONObject(JsonFileName).getJSONObject("Policy");
-                        } catch (Exception e) {
-                        }
+                ToRestApi rest = new ToRestApi();
+                JSONObject obj;
+                int uploadsAccepted=0, uploadsRejected=0, uploadFailed=0;
+                for (int i = 0; i < jsonFiles.length; i++) {
+                    String jsonText = global.getFileText(jsonFiles[i].getPath());
 
-                        resInt = Integer.parseInt(rest.postObjectToRestApiObjectToken(obj, "policy/renew"));
+                    HttpResponse response = null;
+                    int responseCode=-1;
+                    int uploadStatus=-1;
 
-                        if (resInt != 2) serv = true;
-                        else serv = false;
-                    } else {
-                        try {
-                            obj = new JSONObject(JsonFileName).getJSONObject("feedback");
-                        } catch (Exception e) {
+                    try {
+                        obj = new JSONObject(jsonText).getJSONObject(jsonPropertyName);
+                        response = rest.postToRestApiToken(obj,functionName);
+                        if(response!=null) {
+                            responseCode = response.getStatusLine().getStatusCode();
+                            if (responseCode == 200) {
+                                uploadStatus = Integer.parseInt(EntityUtils.toString(response.getEntity()));
+                            }
                         }
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
 
-                        try {
-                            resInt = Integer.parseInt(rest.postObjectToRestApiObjectToken(obj, "feedback"));
-                        } catch (Exception e) {
+                    if(response == null || responseCode!=200 || uploadStatus==2)
```

Comment:
What does uploadStatus==2 mean? Please create a const list with values (1, 2...) and use that instead of actual value.